src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs

```diff
@@ -0,0 +1,373 @@
+            {
+                _fileTraceWriter = NullTraceWriter.Instance;
+            }
+        }
+
+        public static string[] DefaultAssemblyReferences
+            => new string[]
+            {
+                "System",
+                "System.Core",
+                "System.Xml",
+                "System.Net.Http",
+                "Microsoft.WindowsAzure.Storage"
+            };
+
+        public static Assembly[] DefaultAssemblies
+            => new Assembly[]
+            {
+                typeof(TraceWriter).Assembly,
+                typeof(object).Assembly,
+                typeof(TimerInfo).Assembly
+            };
+
+        private FileSystemWatcher InitializeFileWatcher(ScriptHost host)
+        {
+            FileSystemWatcher fileWatcher = null;
+            if (host.ScriptConfig.FileWatchingEnabled)
+            {
+                string functionDirectory = Path.GetDirectoryName(_functionMetadata.Source);
+                fileWatcher = new FileSystemWatcher(functionDirectory, "*.*")
+                {
+                    IncludeSubdirectories = true,
+                    EnableRaisingEvents = true
+                };
+                fileWatcher.Changed += OnScriptFileChanged;
+                fileWatcher.Created += OnScriptFileChanged;
+                fileWatcher.Deleted += OnScriptFileChanged;
+                fileWatcher.Renamed += OnScriptFileChanged;
+            }
+
+            return fileWatcher;
+        }
+
+        private void OnScriptFileChanged(object sender, FileSystemEventArgs e)
+        {
+            // The ScriptHost is already monitoring for changes to function.json, so we skip those
+            string fileName = Path.GetFileName(e.Name);
+            if (string.Compare(fileName, "function.json") != 0)
+            {
+                // Reset cached function
+                _function = null;
+                _fileTraceWriter.Verbose(string.Format("Script for function '{0}' changed. Reloading.", _functionMetadata.Name));
+
+                _fileTraceWriter.Verbose("Compiling function script.");
+                Script<object> script = CreateScript();
+                ImmutableArray<Diagnostic> compilationResult = script.Compile();
+                
+                foreach (var diagnostic in compilationResult)
+                {
+                    var traceEvent = new TraceEvent(GetTraceLevelFromDiagnostic(diagnostic), diagnostic.ToString());
+                    _fileTraceWriter.Trace(traceEvent);
+                }
+            }
+        }
+
+        public override async Task Invoke(object[] parameters)
+        {
+            object input = parameters[0];
+            TraceWriter traceWriter = (TraceWriter)parameters[1];
+            IBinder binder = (IBinder)parameters[2];
+            ExecutionContext functionExecutionContext = (ExecutionContext)parameters[3];
+
+            try
+            {
+                _fileTraceWriter.Verbose(string.Format("Function started"));
```

Comment:
I believe this was in the original code and I didn't bother changing. I'll make that change.